generate.go

```diff
@@ -237,132 +240,121 @@ func Generate(ctxName, cogPath string, outputType Format, filter EnvFilter) (Cfg
 
 func generate(ctxName string, tree *toml.Tree, gear Resolver) (CfgMap, error) {
 	var err error
+	var ctx baseContext
 
-	type rawManifest struct {
-		table map[string]CfgMap
-	}
-
-	// grab manifest name
 	name, ok := tree.Get("name").(string)
-	if !ok || name == "" {
-		return nil, fmt.Errorf("manifest.name string value must be present as a string")
-	}
-	if err = tree.Delete("name"); err != nil {
-		return nil, err
+	if !ok {
+		return nil, fmt.Errorf("manifest.name string value must be present as a non-empty string")
 	}
-
 	gear.SetName(name)
 
-	var manifest rawManifest
-	if err = tree.Unmarshal(&manifest.table); err != nil {
+	ctxTree, ok := tree.Get(ctxName).(*toml.Tree)
+	if !ok {
+		return nil, fmt.Errorf("%s context missing from cog file", ctxName)
+	}
+
+	var ctxMap map[string]interface{}
+	if err := ctxTree.Unmarshal(&ctxMap); err != nil {
 		return nil, err
 	}
 
-	env, ok := manifest.table[ctxName]
-	if !ok {
-		return nil, fmt.Errorf("%s context missing from cog file", ctxName)
+	if err = mapstructure.Decode(ctxMap, &ctx); err != nil {
+		return nil, fmt.Errorf("generate context: %w", err)
 	}
 
-	genOut, err := gear.ResolveMap(env)
+	genOut, err := gear.ResolveMap(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("%s: %w", ctxName, err)
 	}
 
 	return genOut, nil
 }
 
-// parseEnv traverses an map interface to populate a gear's configMap
-func parseEnv(env CfgMap) (linkMap LinkMap, err error) {
+// parseCtx traverses an map interface to populate a gear's configMap
+func parseCtx(ctx baseContext) (linkMap LinkMap, err error) {
 	linkMap = make(map[string]*Link)
 
 	// skip fetching encrypted vars if flag is toggled
 	if !NoEnc {
-		err = decodeEncrypted(linkMap, env)
+		err = decodeEncVars(linkMap, ctx.Enc)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	err = decodeEnv(linkMap, env)
+	err = decodeVars(linkMap, ctx.toContext())
 	if err != nil {
 		return nil, err
 	}
 	return linkMap, nil
 }
 
-func decodeEnv(linkMap LinkMap, env CfgMap) error {
+// baseContext is the struct that maps to the TOML table's ctx name
+type baseContext struct {
```

Comment:
I thought about that a lot but I really think that the inherent meaning of the word here supersedes the namespace confusion it might cause. "Context" in terms of what you want the output of key value store to be is really important.